src/components/symptom-shark/presentational/SymptomSeverityChart/SymptomSeverityChart.tsx

```diff
@@ -0,0 +1,178 @@
+    intervalStart?: Date;
+    symptom: SymptomConfiguration;
+    showAllDays?: boolean;
+}
+
+export default function (props: SymptomSeverityChartProps) {
+    let visualizationContext = useContext(SymptomSharkVisualizationContext);
+    if (!visualizationContext) {
+        return <TextBlock>Error: Symptom Severity Chart must be used inside a Symptom Shark Visualization Coordinator.</TextBlock>
+    }
+    let { logEntries } = visualizationContext;
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let intervalStart = props.intervalStart || startOfMonth(new Date());
+    if (dateRangeContext) {
+        intervalStart = dateRangeContext.intervalStart;
+    }
+
+    var monthDays = getDatesForMonth(intervalStart.getFullYear(), intervalStart.getMonth());
+
+    var symptomAverage = function () {
+        var total = 0;
+
+        var relevantEntries = [];
+        for (var i = 0; i < monthDays.length; i++) {
+            var day = getDayKey(monthDays[i]);
+            if (logEntries[day]) {
+                var matchingSymptom = logEntries[day].symptoms.find((s) => s.id == props.symptom.id);
+                if (matchingSymptom && matchingSymptom.severity) {
+                    relevantEntries.push(matchingSymptom);
+                }
+            }
+        }
+
+        if (props.symptom.severityTracking == '3PointScale') {
+            total += relevantEntries.filter((s) => s.severity && s.severity < 4).length;
+            total += 2 * relevantEntries.filter((s) => s.severity && s.severity >= 4 && s.severity < 8).length;
+            total += 3 * relevantEntries.filter((s) => s.severity && s.severity >= 8).length;
+        }
+        else {
+            for (var j = 0; j < relevantEntries.length; j++) {
+                total += relevantEntries[j].severity ?? 0;
+            }
+        }
+        return total / relevantEntries.length;
+    }
+
+    var symptomSeverity = function (date: Date) {
+        var day = getDayKey(date);
+        if (logEntries[day]) {
+            var matchingSymptom = logEntries[day].symptoms.find((s) => props.symptom.id == s.id);
+            if (matchingSymptom && matchingSymptom.severity) {
+                if (props.symptom.severityTracking == '3PointScale') {
+                    if (matchingSymptom.severity < 4) { return 1; }
+                    if (matchingSymptom.severity >= 4 && matchingSymptom.severity < 8) { return 2; }
+                    return 3;
+                } else {
+                    return matchingSymptom.severity;
+                }
+            }
+        }
+        return 0;
+    }
+
+    var daysMatchingScore = function (score: number) {
+        var result = 0;
+        for (var i = 0; i < monthDays.length; i++) {
+            var severity = symptomSeverity(monthDays[i]);
+            if (severity == score) {
+                result++;
+            }
+        }
+        return result;
+    }
+
+    if (isNaN(symptomAverage())) {
```

Comment:
May want to pull the value from `symptomAverage()` into a variable.  It looks like it gets computed 3 times when rendering.